src/factories/custom/Restricted_PIM_Factory_v1.sol

```diff
@@ -0,0 +1,365 @@
+                        .BondingCurveProperties,
+                    address
+                )
+            );
+        }
+        // set newly deployed mint wrapper as issuance token
+        fundingManagerConfig.configData =
+            abi.encode(address(mintWrapper), bcProperties, collateralToken);
+
+        // deploy workflow
+        IOrchestrator_v1 orchestrator = IOrchestratorFactory_v1(
+            orchestratorFactory
+        ).createOrchestrator(
+            workflowConfig,
+            fundingManagerConfig,
+            authorizerConfig,
+            paymentProcessorConfig,
+            moduleConfigs
+        );
+
+        // return deployed contracts and all relevant decoded config params
+        return (
+            DeployedContracts({
+                issuanceToken: issuanceToken,
+                mintWrapper: mintWrapper,
+                orchestrator: orchestrator
+            }),
+            DecodedConfigParams({
+                collateralToken: IERC20(collateralToken),
+                initialCollateralSupply: bcProperties.initialCollateralSupply,
+                initialIssuanceSupply: bcProperties.initialIssuanceSupply,
+                realAdmin: realAdmin
+            })
+        );
+    }
+
+    function _configureWorkflow(
+        DeployedContracts memory deployedContracts,
+        DecodedConfigParams memory decodedConfigParams,
+        address beneficiary
+    ) private {
+        // get bonding curve / funding manager
+        address fundingManager =
+            address(deployedContracts.orchestrator.fundingManager());
+        // get payment router
+        address[] memory modules = deployedContracts.orchestrator.listModules();
+        address paymentRouter;
+        bytes4 paymentRouterInterfaceId =
+            type(ILM_PC_PaymentRouter_v1).interfaceId;
+        for (uint i; i < modules.length; ++i) {
+            if (
+                ERC165(modules[i]).supportsInterface(
+                    paymentRouterInterfaceId
+                )
+            ) {
+                paymentRouter = modules[i];
+            }
+        }
+
+        // enable bonding curve to mint issuance token
+        deployedContracts.mintWrapper.setMinter(fundingManager, true);
+
+        // transfer initial collateral supply from realAdmin to bonding curve and mint issuance token to msg.sender
+        _manageInitialSupplies(
+            IBondingCurveBase_v1(fundingManager),
+            decodedConfigParams.collateralToken,
+            deployedContracts.issuanceToken,
+            decodedConfigParams.initialCollateralSupply,
+            decodedConfigParams.initialIssuanceSupply,
+            decodedConfigParams.realAdmin,
+            beneficiary
+        );
+
+        // assign permissions to buy/sell from curve to admin
+        bytes32 curveAccess = FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1(
+            fundingManager
+        ).CURVE_INTERACTION_ROLE();
```

Comment:
Same as here: https://github.com/InverterNetwork/contracts/pull/672#discussion_r1808532724

Since the motivation of this factory is foremost a highly use case specific UX improvement, we can live with the assumption that user inputs to the factory are correct.